ctrd/container.go

```diff
@@ -385,7 +385,10 @@ func (c *Client) destroyContainer(ctx context.Context, id string, timeout int64)
 		}
 		msg = waitExit()
 	}
-	if err := msg.RawError(); err != nil && errtypes.IsTimeout(err) {
+
+	// ignore the error is stop time out
+	// TODO: how to design the stop error is time out?
+	if err := msg.RawError(); err != nil && !errtypes.IsTimeout(err) {
```

Comment:
The `Kill` rpc works well and returns nil error during `destroyContainer`. Basically, we can say that all the processes in container are be killed. But the dead event doesn't show up immediately after `Kill` call. It will be reaped by containerd-shim which will set the state to dead. Therefore, the `ProbeContainer` could not get the exit event in time. I think this is tricky thing. The containerd-shim might be hang by dead lock so that there is always timeout.

Basically, LGTM for ignoring the timeout error. @Ace-Tang how do you think?

Reply:
Still confuse me to ignore the timeout error.

If we can confirm the timeout error is ok, how about add a retry action to make sure the containerd-shim really exited?

Reply:
we don't know how many times we need to try, right?

If the containerd-shim doesn't set the dead state, we will get error from `pack.container.Delete(ctx)` because there are still running processes.

Reply:
If we do not know when the shim exited, i do not think to ignore the error is a good solution.

Reply:
@HusterWan send kill signal is ok, retry to send kill signal is unuse.